extractor/docker/docker.go

```diff
@@ -196,14 +235,14 @@ func (d Extractor) Extract(ctx context.Context, imageName string, filenames []st
 	for _, ref := range m.Manifest.Layers {
 		layerIDs = append(layerIDs, string(ref.Digest))
 		go func(dig digest.Digest) {
-			d.extractLayerWorker(dig, r, ctx, image, errCh, layerCh)
+			d.extractLayerWorker(dig, r, ctx, image, errCh, layerCh, filenames)
 		}(ref.Digest)
 	}
 
-	filesInLayers := make(map[string]extractor.FileMap)
-	opqInLayers := make(map[string]extractor.OPQDirs)
+	filesInLayers := map[string]extractor.FileMap{}
+	opqInLayers := map[string]extractor.OPQDirs{}
 	for i := 0; i < len(m.Manifest.Layers); i++ {
-		if err := d.extractLayerFiles(layerCh, errCh, ctx, filenames, filesInLayers, opqInLayers); err != nil {
+		if filesInLayers, opqInLayers, err = d.extractLayerFiles(layerCh, errCh, ctx, filenames); err != nil {
```

Comment:
MUST: `extractLayerFiles` always overwrite `filesInLayers` by accident. This is a cause of a bug. `filesInLayers` must have necessary files for each layer. The current implementation has files only for one layer since it overwrites the variable.

- sha256:976a760c94fcdd7d105269ae621e8269e7bb25a58c52ae667b4029a6bc7e33cb
  - `etc/lsb-release` : content
- sha256:c58992f3c37bb64aeba18910408cda9a7a63e212fe27e95065a8d54130ca5926
  - `etc/os-release` : content

This bug affects all OSes, not just Ubuntu. It reproduces when the image has some layers. The image having only one layer doesn't overwrite the variable, so this bug doesn't reproduce. Coincidentally Ubuntu made us notice this bug 😅 We have to add this case to the unittest.

Reply:
Wonderful observation! Thanks. I've fixed it here: https://github.com/aquasecurity/fanal/pull/51/commits/cad6d3165f66009286fb0548498231cd4017ff94